src/segmantic/prepro/labels.py

```diff
@@ -1,30 +1,54 @@
 import numpy as np
 import colorsys
+from typing import Callable, Dict, List, Tuple
 
 
-def build_map(input_label_names, mapper):
-    input_label_map = {n: i for i, n in enumerate(input_label_names)}
+def build_tissue_mapping(input_label_map: Dict[str, int], mapper: Callable[[str], str]):
+    """build mapping to relable
 
-    output_label_names = list(sorted(set([mapper(n) for n in input_label_names])))
+    Args:
+        input_label_map (Dict[str, int]): list of tissue names, e.g. loaded with 'load_tissue_list'
+        mapper (Callable[[str], str]): function that maps a tissue name to a new name
+
+    Returns:
+        Dict[str, int]: tissue label dict after mapping
+        np.ndarray: input-to-output lookup table
+    """
+    output_label_names = list(sorted(set([mapper(n) for n in input_label_map.keys()])))
     output_label_names.remove("Background")
     output_label_names = ["Background"] + output_label_names
     output_label_map = {n: i for i, n in enumerate(output_label_names)}
 
-    input2output = np.zeros((len(input_label_names),), dtype=np.uint16)
+    input2output = np.zeros((len(input_label_map),), dtype=np.uint16)
     for name in input_label_map.keys():
         index = input_label_map[name]
         name_mapped = mapper(name)
         index_mapped = output_label_map[name_mapped]
         input2output[index] = index_mapped
-    return input_label_map, output_label_map, input2output
+    return output_label_map, input2output
 
 
-def save_tissue_list(output_label_map:dict, tissue_list_file_name:str):
-    with open(tissue_list_file_name, "w") as f:
-        num_tissues = max(output_label_map.values())
-        label_tissue_map = {}
-        for n in output_label_map.keys():
-            label_tissue_map[output_label_map[n]] = n
+def save_tissue_list(
+    tissue_label_map: Dict[str, int],
+    tissue_list_file_name: str,
+    tissue_color_map: Callable[[str], Tuple[float, float, float]] = None,
```

Comment:
TIP:  ``tissue_color_map`` is a callable that takes a string and returns  RGB tuple
what about ```Optional[ Callable[[str], RGBTuple] ] ```?
where
```python
RGBTuple = Tuple[float, float, float]] 
# or
import collections
RGBTuple = collections.namedtuple("RGBTuple", "r g b")

```


``Tuple[float, float, float]``